File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int f = -1, second = -1;
        int n = nums.length;
        int l = 0;
        int r = n ;

        while (l < r) {
            int m = (l + r) / 2;
            if (nums[m] == target)
                second = m;
            if (nums[m] <= target)
                l = m + 1;
            else
                r = m;
        }

        l = 0;
        r = n ;
        while (l < r) {
            int m = (l + r) / 2;
            if (nums[m] == target)
                f = m;
            if (nums[m] >= target)
                r = m;
            else
                l = m + 1;
        }

        return new int[] { f, second };

    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
import bisect


class Solution:
    def searchRange(self, arr: List[int], x: int) -> List[int]:
        n = len(arr)
        if n == 0:
            return [-1, -1]
        index = bisect.bisect_left(arr, x, 0, n)
        index_right = bisect.bisect_right(arr, x, 0, n) - 1

        if index < n and arr[index] != x:
            index = -1
        if index_right == n:
            index_right = -1
        if index == n:
            index = -1

        if index_right < n and arr[index_right] != x:
            index_right = -1
        return [index, index_right]
